### PR TITLE
create hidden field

### PR DESCRIPTION
**Pull request by weili-govtech**
**Changes**
- new `hidden-field` that renders hidden input
- comes with validation and conditional rendering but no label
- delete branch

### CODE REVIEW

File: src/components/fields/index.ts [hunk 1]
```diff
@@ -4,8 +4,9 @@ export * from "./chips";
 export * from "./contact-field";
 export * from "./date-field";
 export * from "./date-range-field";
-export * from "./file-upload";
+export * from "./hidden-field";
 export * from "./histogram-slider";
+export * from "./file-upload";
```
Comment (qroll) [inline src/components/fields/index.ts:9]: was moving `file-upload` intended? as it was already in the correct alphabetical order

File: src/components/fields/hidden-field/hidden-field.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+		schema: { showIf: _showIf, uiType: _uiType, validation, ...otherSchema },
+		...otherProps
+	} = props;
+	const { setFieldValidationConfig } = useValidationConfig();
+
+	// =============================================================================
+	// EFFECTS
+	// =============================================================================
+	useEffect(() => {
+		setFieldValidationConfig(id, Yup.string(), validation);
```
Comment (qroll) [inline src/components/fields/hidden-field/hidden-field.tsx:22]: something to consider as a future enhancement is the ability to handle other types. a pain point I've encountered is having to cast boolean to string.

File: src/components/fields/index.ts (hunk 1 above)
Comment (weili-govtech) [reply src/components/fields/index.ts]: whoops not intended, I have fixed it

File: src/components/fields/hidden-field/hidden-field.tsx (hunk 2 above)
Comment (weili-govtech) [reply src/components/fields/hidden-field/hidden-field.tsx:22]: oh yeah thanks for the suggestion, I will introduce a new validation rule for hidden field to validate as other data types

**Review by qroll — APPROVED** [review]

**Pull request merged**